src/System.Diagnostics.PerformanceCounter/ref/System.Diagnostics.PerformanceCounter.cs

```diff
@@ -205,3 +205,88 @@ public enum PerformanceCounterType
         Timer100NsInverse = 558957824,
     }
 }
+namespace System.Diagnostics.PerformanceData
+{
+    public sealed class CounterData
+    {
+        unsafe internal CounterData(long* pCounterData) { }
+        public void Decrement() { throw null; }
+        public void Increment() { throw null; }
+        public void IncrementBy(long value) { throw null; }
+        public long RawValue { get { throw null; } set { } }
+        public long Value { get { throw null; } set { } }
+    }
+    public class CounterSet : IDisposable
+    {
+        ~CounterSet() { throw null; }
+        public void AddCounter(int counterId, CounterType counterType) { throw null; }
+        public void AddCounter(int counterId, CounterType counterType, string counterName) { throw null; }
+        public CounterSet(Guid providerGuid, Guid counterSetGuid, CounterSetInstanceType instanceType) { }
+        public CounterSetInstance CreateCounterSetInstance(string instanceName) { throw null; }
+        public void Dispose() { throw null; }
+        protected virtual void Dispose(bool disposing) { throw null; }
+    }
+    public sealed class CounterSetInstance : IDisposable
+    {
+        ~CounterSetInstance() { throw null; }
+        internal CounterSetInstance(CounterSet counterSetDefined, string instanceName) { }
+        public void Dispose() { throw null; }
+        public CounterSetInstanceCounterDataSet Counters { get { throw null; } }
+    }
+    public sealed class CounterSetInstanceCounterDataSet : IDisposable
+    {
+        ~CounterSetInstanceCounterDataSet() { throw null; }
+        internal CounterSetInstanceCounterDataSet(CounterSetInstance thisInst) { }
+        public void Dispose() { throw null; }
+        public CounterData this[int counterId] { get { throw null; } }
+        public CounterData this[string counterName] { get { throw null; } }
+    }
+    public enum CounterType
```

Comment:
How different are these from the ones listed in PerformanceCounterType.cs? Can we merge them?

Reply:
They are in different namespaces 
`System.Diagnostics.PerformanceData.CounterType` https://referencesource.microsoft.com/#System.Core/System/Diagnostics/PerformanceData/CounterType.cs
`System.Diagnostics.PerformanceCounterType` https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/PerformanceCounterType.cs,dbfac66e4bef3702

and some counters are not present.
I apologize maybe i don't understand well what you mean :worried:

Reply:
> If they are different, then having the two different enums in different namespaces is fine.

i did some check

|PerformanceCounterType.cs|PerformanceData/CounterType.cs||
|----------|:-------------:|------:|
|SampleFraction|SampleFraction|same|
|SampleCounter|SampleCounter|same|
|SampleBase|SampleBase|same|
|AverageTimer32|AverageTimer32|same|
|AverageBase|AverageBase|same|
|AverageCount64|AverageCount64|same|
|ElapsedTime|ElapsedTime|same|
|RateOfCountsPerSecond32|RateOfCountsPerSecond32|same|
|RateOfCountsPerSecond64|RateOfCountsPerSecond64|same|
|CountPerTimeInterval32|QueueLength|different name|
|CountPerTimeInterval64|LargeQueueLength|different name|
|NumberOfItems32|RawData32|differen tname|
|NumberOfItems64|RawData64|differentname|
|NumberOfItemsHEX32|RawDataHex32|different name|
|NumberOfItemsHEX64|RawDataHex64|different name|
|RawFraction|RawFraction32|different name|
|RawBase|RawBase32|different name|
|CounterTimer|PercentageActive|different name|
|CounterTimerInverse|PercentageNotActive|different name|
|Timer100Ns|PercentageActive100Ns|different name|
|Timer100NsInverse|PercentageNotActive100Ns|differen tname|
|CounterMultiTimer|MultiTimerPercentageActive|different name|
|CounterMultiTimerInverse|MultiTimerPercentageNotActive|different name|
|CounterMultiTimer100Ns|MultiTimerPercentageActive100Ns|different name|
|CounterMultiTimer100NsInverse|MultiTimerPercentageNotActive100Ns|different name|
|CounterMultiBase|MultiTimerBase|different name|
|CounterDelta32|Delta32|different name|
|CounterDelta64|Delta64|different name|
||ObjectSpecificTimer|missing|
||PrecisionSystemTimer|missing|
||PrecisionTimer100Ns|missing|
||PrecisionObjectSpecificTimer|missing|
||QueueLength100Ns|missing|
||QueueLengthObjectTime|missing|
||RawFraction64|missing|
||RawBase64|missing|

They differ for name or missing counter.

> then we should share them between the two, rather than have two copies, and put them in Common

How we can share enums values between two different enum name?

Reply:
>> Maybe a stupid question...how we can share enums values between two different enum name?

changing the definitions to use any of one enum should work.

Reply:
> changing the definitions to use any of one enum should work.

Mmm...can you provide an example?I'm not use to understand sorry

Reply:
Try to explain better, do you mean:

on PerformanceData.CounterType.cs
```cs
 SampleFraction = PerformanceCounterType.SampleFraction,
```
or
```cs
 SampleFraction = Interop.Kernel32.PerformanceCounterOptions.PERF_SAMPLE_FRACTION,
```
it's not clear to me `we should share them between the two, rather than have two copies, and put them in Common`

Reply:
My thought was that we'd use what is already in Common, rather than have them explicitly defined in both classes.

Reply:
I agree with you to avoid duplications...but i don't understand what you mean when you say "already in Common". Is there a file with enums definition already on "Common folder"?
If i search for AverageBase for instance i don't found anything on Common folder https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=AverageBase+&type=
Or you meant create a file with 2 enums on common folder and add to src/ref projects to avoid duplication...is this your idea?

Reply:
Ah ok. So what I was thinking is that you'd use https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Windows/kernel32/Interop.PerformanceCounterOptions.cs like we do in PerformanceCounterType. So AverageBase = Interop.Kernel32.PerformanceCounterOptions.PERF_AVERAGE_BASE, etc.

Reply:
Ok so "simply" reuse Interop.Kernel32.PerformanceCounterOptions!

Reply:
done